Bottom: 6fa32942fcab249f67d898448bcefa6ce68982f5
Top:    a3a637e86833c720d0d1bf2c609bfd6cf0aa3575
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-05 18:49:24 +0800

autofs-5.0.7 - fix memory leak in get_dc_list()

In get_dc_list() if an invalid port is found the allocated list storage
isn't freed on the error exit.


---

diff --git a/modules/dclist.c b/modules/dclist.c
index d16b913..af21ce0 100644
--- a/modules/dclist.c
+++ b/modules/dclist.c
@@ -536,6 +536,7 @@ struct dclist *get_dc_list(unsigned int logopt, const char *uri)
 		else
 			strcat(tmp, " ");
 
+		list = NULL;
 		for (i = 0; i < numdcs; i++) {
 			if (i > 0)
 				strcat(tmp, " ");
@@ -549,6 +550,7 @@ struct dclist *get_dc_list(unsigned int logopt, const char *uri)
 					error(logopt,
 					      "invalid port: %u", dcs[i].port);
 					free_srv_rrs(dcs, numdcs);
+					free(tmp);
 					goto out_error;
 				}
 				strcat(tmp, port);
